src/lib.rs

```diff
@@ -324,24 +338,21 @@ fn num_chars_generic<Chunk: ByteChunk<Splat = Chunk>>(naive_below: usize,
                                                     -> usize {
     // Extract pre/post so naive_count is only inlined once.
     let len = haystack.len();
-    let mut count = len;
+    let mut count = 0;
     let unchunked = if len < naive_below {
         [haystack, &haystack[0..0]]
-    } else if len > group_above {
-        let (pre, mid, post) = chunk_align::<[Chunk; 4]>(haystack);
-        count -= chunk_num_chars(mid);
+    } else if len <= group_above {
```

Comment:
nit: If you're switching this condition around, you should probably do the same to the other to keep them in sync.